iris-mpc-cpu/src/protocol/shared_iris.rs

```diff
@@ -50,26 +50,52 @@ impl GaloisRingSharedIris {
         })
     }
 
-    /// Generate iris code shares of an input iris code using local randomness
+    /// Generate iris code shares of an input iris code using local randomness, alongside with
+    /// its mirrored component
     pub fn generate_shares_locally<R: Rng + CryptoRng>(
         rng: &mut R,
         iris: IrisCode,
-    ) -> [GaloisRingSharedIris; 3] {
+    ) -> ([GaloisRingSharedIris; 3], [GaloisRingSharedIris; 3]) {
         let code_shares = GaloisRingIrisCodeShare::encode_iris_code(&iris.code, &iris.mask, rng);
         let mask_shares = GaloisRingIrisCodeShare::encode_mask_code(&iris.mask, rng);
-        [
-            GaloisRingSharedIris {
-                code: code_shares[0].clone(),
-                mask: GaloisRingTrimmedMaskCodeShare::from(&mask_shares[0]),
-            },
-            GaloisRingSharedIris {
-                code: code_shares[1].clone(),
-                mask: GaloisRingTrimmedMaskCodeShare::from(&mask_shares[1]),
-            },
-            GaloisRingSharedIris {
-                code: code_shares[2].clone(),
-                mask: GaloisRingTrimmedMaskCodeShare::from(&mask_shares[2]),
-            },
-        ]
+        let code_shares_mirrored = code_shares
```

Comment:
@dkales I don't particularly love this change, as it requires changes to a lot of caller code. Do you think there's a way around it, or a simple refactor to add a more descriptive type as the output of this function makes more sense?

Reply:
I guess @naure should have the say here on if this is needed/useful in the test.

Reply:
Alternatively, this function could stay unchanged, and move the conversion to mirrored version in a separate function.

Reply:
@naure I have adopted your suggestion, it is indeed quite cleaner!